Sources/Navidux/Core/Navigation/NavigationAction/Actions/RestructNavigationAction.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  RestructNavigationAction.swift
+//
+//
+//  Created by Tamerlan Satualdypov on 19.02.2024.
+//
+
+import UIKit
+
+public struct RestructNavigationAction: NavigationAction {
+    public enum AnimationType {
+        case forward
+        case backward
+    }
+    
+    private let modules: [Module]
+    private let animationType: AnimationType
+    
+    init(modules: [Module], animationType: AnimationType) {
+        self.modules = modules
+        self.animationType = animationType
+    }
+    
+    public func perform(on coordinator: Coordinator) {
+        let stack = self.modules.map { $0.assembly(using: coordinator) as UIViewController }
```

Comment:
here we sholud stay in navidux paradigm and use `NavigationScreen` instead of `UIViewController`. and then pass `[NavigationScreen]` in `update` methods where it will be resloved in `[UIViewController]` to update `navigationController` property in `Coordindator`

Reply:
@darkkifir @Bananzaaa @baluhman need your help here

Reply:
Agree with abstraction, in actions we should use `NavigationScreen`, not `UIResponder` child

Reply:
I agree with you, but if we want to force usage of `NavigationScreen` we need to refactor the API of `NavigationController` to use `NavigationScreen` in first place. I have not done it in this PR, but definitely will work on it in the next iteration

Reply:
ok. looks like a new issue😀